### DISCUSSION TITLE
Can't enable dkim_signing

### DISCUSSION THREAD

**Discussion by jaysee** [rspamd/rspamd#4147] [Q&A]
Hello,

I just realize that migrating confiuration from rmilter to dkim_signing was not working, I can't understant whats wrong

Configuration seems OK and  `rspamadm configdump dkim_signing` reports
```config
*** Section dkim_signing ***
use_esld = true;
selector = "default";
use_redis = false;
key_prefix = "DKIM_KEYS";
path = "/etc/dkim/$domain.$selector.key";
check_pubkey = false;
allow_pubkey_mismatch = true;
use_domain = "header";
allow_hdrfrom_mismatch = false;
enabled = true;
allow_username_mismatch = false;
domain {
    example.com {
        path = "/var/lib/rspamd/dkim/example.key";
        selector = "ds";
    }
}
try_fallback = true;
allow_envfrom_empty = true;
allow_hdrfrom_multiple = false;
symbol = "DKIM_SIGNED";
sign_local = true;
sign_authenticated = true;

*** End of section dkim_signing ***
```

Enabled modules via `rspamadm configdump -m` does not show the module : 
`Modules enabled: fuzzy_check, regexp, chartable, dkim`

It indeed doesn't shows a lot of others enabled modules (arc, antivirus, greylisting, ...)

I use postfix with `smtpd_milters = inet:127.0.0.1:11332` in `main.cf`

I tryed enabling debug for this module but nothing related for `dkim_signing` in the logs

What can be wrong ?

regards

**Comment by koehn** [comment]
You need to enable outbound miltering from postfix in `main.cf`:
non_smtpd_milters=inet:127.0.0.1:11332 # For invocation via LDA

See [here](https://tonygravagno.github.io/rspamd.com/doc/tutorials/scanning_outbound.html).

**Comment by jaysee** [comment]
hi, thanks,

but I already tried, DKIM signing is not done...

I stick to opendkim to do the signing, but would prefere to avoid another service...

**Reply by fatalbanana** [reply]
> I tryed enabling debug for this module but nothing related for dkim_signing in the logs

If you've really `debug_modules = ["dkim_signing"];` set in `local.d/logging.inc` or equivalent you should probably see something. Reason you might not could be that `settings` are in use `rspamd_task_write_log` line should give that away if it's so.

**Comment by jaysee** [comment]
yep, i got some logging :
`dkim_symbol_callback: skip DKIM checks for local networks and authorized users(rspamd_proxy)`

but can't understand as i got 
```
sign_authenticated = true;
sign_local = true;
```

**Reply by fatalbanana** [reply]
> `dkim_symbol_callback: skip DKIM checks for local networks and authorized users(rspamd_proxy)`

This log is about DKIM checking, `dkim_signing` logs say `dkim_signing` in them.

```2023-10-30 10:27:50 #2581343(controller) <27193b>; csession; rspamd_task_write_log: id: ...```

The hash after the worker name (`27193b`) can be used to find logs related to a particular `task` or message.

**Reply by koehn** [reply]
That logging is from the code that _verifies_ an existing DKIM signature (which it is skipping, correctly, because the user is authenticated), not the `dkim_signing` module that _produces_ a signature.

**Comment by jaysee** [comment]
mm yes... nothing about `dkim_signing`...
